### DISCUSSION TITLE
Does Subscribing work after offline?

### DISCUSSION THREAD

**Discussion by richard457** [pocketbase/pocketbase#2119] [Q&A]
I am building a distributed system where one system can update or create data while another device is offline, I checked when the offline device comes back online it does not listen or subscribe to the change, is this expected behavior that both devices need to be online to receive real-time data?

**Comment by ganigeorgiev** [comment]
The realtime events are not stored anywhere. The subscription works only with events that are received **after** the realtime connection was established.

If you want to fetch "older" events, you'll have to perform a regular `getOne`/`getList` request before the `subscribe` call.

**Reply by richard457** [reply]
I can see that RecordModel does not have a delete event, also how deep does it return the event or does every data recorded has an event associated regardless of how old it is?

**Reply by richard457** [reply]
in your suggestion then it can be good to not make subscribe calls at all so it is guaranteed that these events are there every time?

**Reply by ganigeorgiev** [reply]
I don't understand what do you mean. Once again, the realtime events are not stored anywhere.

The `subscribe()` callback reacts only to changes. It doesn't perform fetch.

When you create/update/delete a record we broadcast that action to all currently available subscribers that are allowed to list/view the record.

**Reply by richard457** [reply]
I got confused because you said that I can fetch older "events" using getOne/getList before subscribing can give me events, anyway seems like getOne/getList will not give me events after a while as events are not stored anywhere, any plans on saving events to be fetched later on?

**Reply by ganigeorgiev** [reply]
> any plans on saving events to be fetched later on?

This is more of a client-side/SDK issue since in order to mark an event as "new/created" you have to compare it and/or have information against some previous client state.

Local/offline DB sync for the JS and Dart SDKs is in my todo but for now is not a priority and it is left to the developers to handle it on their own if they need it.

**Reply by richard457** [reply]
Noted, Let me think of possible implementation, I think there are a lot of people like me who will love to have Local/offline DB sync and since is on your todo, we are patiently waiting.
